spec/lib/nerve/reporter_zookeeper_spec.rb

```diff
@@ -29,5 +32,109 @@
     reporter.report_up
     reporter.stop
   end
+
+  context "when reporter is up" do
+    before(:each) do
+      allow(zk).to receive(:close!)
+      allow(zk).to receive(:connected?).and_return(true)
+      allow(zk).to receive(:exists?) { "zk_path" }.and_return(false)
+      allow(zk).to receive(:mkdir_p) { "zk_path" }
+      allow(zk).to receive(:create) { "full_path" }
+      allow(zk).to receive(:set) { "full_path" }
+      allow(zk).to receive(:delete).with("full_path", anything())
+      allow(ZK).to receive(:new).and_return(zk)
+      @reporter = Nerve::Reporter::Zookeeper.new(subject)
+      @reporter.start
+      @reporter.report_up
+    end
+
+    after(:each) do
+      # reset the class variable to avoid mock object zk leak
+      Nerve::Reporter::Zookeeper.class_variable_set(:@@zk_pool, {})
+      Nerve::Reporter::Zookeeper.class_variable_set(:@@zk_pool_count, {})
+    end
+
+    it "returns true on report_up" do
+      expect(@reporter.report_up).to be true
+    end
+
+    it "returns true on report_down" do
+      expect(@reporter.report_down).to be true
+    end
+
+    it "returns true on ping?" do
+      expect(zk).to receive(:exists?) { "zk_path" }.and_return(true)
+      expect(@reporter.ping?).to be true
+    end
+
+    context "when zk.connected? started to return false" do
+      it 'returns false on ping?' do
+        # this condition is triggered if connection has been lost for a while (a few sec)
+        expect(zk).to receive(:connected?).and_return(false)
```

Comment:
move this expectation out into the context block and out of the example blocks:

```ruby
context "when zk.connected? begins to return false" do
  before { expect(zk).to receive(:connected?).and_return(false)

  # examples go here
end
```

Reply:
done